javascript-local-storage/data.js

```diff
@@ -0,0 +1,12 @@
+/* exported todos */
+
+var todos = [];
+
+window.addEventListener('beforeunload', function (event) {
+  var todosJSON = JSON.stringify(todos);
+  localStorage.setItem('javascript-local-storage', todosJSON);
+});
+
+var previousTodosJSON = localStorage.getItem('javascript-local-storage');
+
+todos = JSON.parse(previousTodosJSON);
```

Comment:
You'll want to use a conditional to check if `getItem` returned `null` or not. You can test that you code works properly by opening an In Cognito / Private Browsing window.